scripts/BilbyCustomFunctions_Reduction.py

```diff
@@ -0,0 +1,343 @@
+    """  Strip NaNs from the 1D OutputWorkspace """  # add isinf
+
+    data = output_workspace.readY(0)
+    start_index = next((index for index in range(len(data)) if not math.isnan(data[index])), None)
+    end_index = next((index for index in range(len(data) - 1, -1, -1) if not math.isnan(data[index])), None)
+
+    q_values = output_workspace.readX(0)
+    start_q = q_values[start_index]
+    end_q = q_values[end_index]
+
+    CropWorkspace(InputWorkspace=output_workspace, XMin=start_q, XMax=end_q, OutputWorkspace=base_output_name)
+
+    return base_output_name
+
+
+#######################################################################################
+# GENERAL #############################################################################
+#######################################################################################
+
+def GetPixelSize():  # reads current IDF and get pixelsize from there
+    """ To get pixel size for Bilby detectors from the Bilby_Definition.xml file """
+
+    from mantid.api import ExperimentInfo
+    import xml.etree.cElementTree as ET
+
+    currentIDF = ExperimentInfo.getInstrumentFilename("Bilby")
+    # print currentIDF
+    tree = ET.parse(currentIDF)
+    for node in tree.iter():
+        if node.tag == "{http://www.mantidproject.org/IDF/1.0}height":
+            name = node.attrib.get('val')
+            break
+    pixelsize = float(name)
+
+    return pixelsize
+
+
+#######################################################################################
+
+def ReadCSV(filename):
+    """ Read cvs... """
+
+    parameters = []
+    with open(filename) as csvfile:
+        reader = csv.reader(csvfile)
+        for row in reader:
+            parameters.append(row)
+    return parameters
+
+
+#######################################################################################
+
+def AttenuationCorrection(att_pos, data_before_May_2016):
+    """ Bilby has four attenuators; before May 2016 there were only two.
+     Value of the attenuators are hard coded here and being used for the I(Q) scaling in Q1D """
+
+    if (data_before_May_2016):
+        print("You stated data have been collected before May, 2016, i.e. using old attenuators. Please double check.")
+        if (att_pos == 2.0 or att_pos == 4.0):
+            print(
+                "Wrong attenuators value; Either data have been collected after May, 2016, or something is wrong with hdf file")
+            sys.exit()
+        if att_pos == 1.0:
+            scale = 0.007655  # att 130
+        elif att_pos == 3.0:
+            scale = 1.0
+        elif att_pos == 5.0:
+            scale = 0.005886  # att 170
+    else:
+        if att_pos == 1.0:
+            scale = 1.0
+        elif att_pos == 2.0:
+            scale = 0.00955
+        elif att_pos == 3.0:
+            scale = 0.005886
+        elif att_pos == 4.0:
+            scale = 0.00290
+        elif att_pos == 5.0:
+            scale = 0.00062
+
```

Comment:
For this function would it make sense possibly to have these values in a dictionary? perhaps a global dictionary defined at the top of the file since this is just a script.